tests/python_package_test/test_dask.py

```diff
@@ -172,7 +176,7 @@ def test_classifier(output, centers, client, listen_port):
     assert_eq(p1_local, p2)
     assert_eq(y, p1_local)
 
-    client.close()
+    client.close(timeout=CLIENT_CLOSE_TIMEOUT)
```

Comment:
Maybe we can wrap `client` as a resource-like fixture?

```
@pytest.fixture()
def resource():
    print("setup")
    yield "resource"
    print("teardown")
```
https://stackoverflow.com/a/39401087

It will reduce code duplication and help to avoid cases where we forget to add this code line.
https://github.com/microsoft/LightGBM/issues/3829#issuecomment-769408850

Reply:
The client we import from `distributed` is already itself a `pytest` fixture: https://github.com/dask/distributed/blob/b5c36b587f0e3295fe59db330603d5c78b39331f/distributed/utils_test.py#L543-L547

I'm really nervous about adding a second layer of `@pytest.fixture()` on top of this, when this PR's goal is specifically to fix an error caused by instability in teardowns. I'm worried that will expose us to new types of problems related to having fixtures-of-fixtures.